### PR TITLE
Added ternary operator and unary plus operator

### PR DESCRIPTION
**Pull request by ggetz**
Added ternary conditional and unary plus operators, and also updated the tests.

### CODE REVIEW

File: Specs/Scene/ExpressionSpec.js [hunk 1]
```diff
@@ -524,4 +534,12 @@ defineSuite([
         expression = new Expression(new MockStyleEngine(), 'isFinite(Color("white"))');
         expect(expression.evaluate(undefined)).toEqual(false);
     });
+
+    it('evaluates ternary conditional', function() {
```
Comment (pjcozzi) [inline Specs/Scene/ExpressionSpec.js]: Add tests with more complex expressions, e.g., `(!(1 + 2 > 3)) ? (2 > 1 : 1 + 1 : 0) : (2 > 1 : -1 + -1 : 0)` should be `2`.

File: Source/Scene/Expression.js [hunk 2]
```diff
@@ -261,7 +267,9 @@ define([
     }
 
     function setEvaluateFunction(node) {
-        if (node._type === ExpressionNodeType.BINARY) {
+        if (node._type === ExpressionNodeType.TERNARY) {
```
Comment (pjcozzi) [inline Source/Scene/Expression.js]: Looking forward, from the AST perspective (not the tokenized perspective), perhaps `ExpressionNodeType.TERNARY` is better named `ExpressionNodeType.CONDITIONAL`.

**Comment by ggetz** [comment]
OK, updated the tests. I also realized that some of the error checking with greater than/less than operations do not mimic JS behavior, so I removed it and updated the related specs.

Also changed `TERNARY` to `CONDITIONAL`

**Comment by pjcozzi** [comment]
Did you push the commit?  I don't see it here.

**Comment by ggetz** [comment]
Sorry, updated.

**Comment by pjcozzi** [comment]
:+1:

**Pull request merged**